src/main/java/edu/umich/its/cpm/MigrationTaskService.java

```diff
@@ -947,13 +948,30 @@ public void onProgressChanged(long numBytes,
 	}
 
 	/**
-	 * for Web Link and citation type of resources, append ".html" to the file name String
+	 * If file extension is missing, look up the extension by file MIME type
```

Comment:
This is a bit more general.  It will add missing extensions, not just the html extension.